libvast/src/system/spawn_sink.cpp

```diff
@@ -38,48 +40,61 @@ namespace vast::system {
 
 namespace {
 
-template <class Writer>
+template <class Writer, class Defaults>
 maybe_actor spawn_generic_sink(caf::local_actor* self, spawn_arguments& args) {
   if (!args.empty())
     return unexpected_arguments(args);
-  VAST_UNBOX_VAR(out, detail::make_output_stream(args.options));
+  std::string category = Defaults::category;
+  auto& options = args.options;
+  auto output = get_or(options, category + ".write", Defaults::write);
+  auto uds = get_or(options, category + ".uds", false);
+  VAST_UNBOX_VAR(out, detail::make_output_stream(output, uds));
```

Comment:
This could use the convenience overload of `make_output_stream`.

Reply:
Good catch! Will do.